modules/data-entry/src/main/frontend/src/questionnaire/Subject.jsx

```diff
@@ -466,9 +466,13 @@ function FormData(props) {
     setData([]);  // Prevent an infinite loop if data was not set
   };
 
+  // Fetch this Form's data
+  useEffect(() => {
+    getFormData(formID);
+  }, []);
```

Comment:
To be super safe, I would put `formId` as the condition:

```suggestion
  }, [formID]);
```

Reply:
Implemented with commit `68276fa` :heavy_check_mark: